File: lib/java/src/main/java/uapi/internal/mock/PartiallyMatches.java
package uapi.internal.mock;

import static uapi.internal.mock.IsSubMapEntryEqual.isSubMapEntryEqual;

import java.util.List;

public class PartiallyMatches {
    public static boolean partiallyMatches(List<Object> wholeList, Object partElement) {
        for (final var wholeElement : wholeList) {
            if (isSubMapEntryEqual(partElement, wholeElement)) {
                return true;
            }
        }

        return false;
    }
}

File: lib/py/uapi/internal/mock/PartiallyMatches.py
def partially_matches(whole_list: list[object], part_element: object) -> bool:
    from uapi.internal.mock.IsSubMapEntryEqual import is_sub_map_entry_equal

    for whole_element in whole_list:
        if is_sub_map_entry_equal(part_element, whole_element):
            return True
    return False
